src/ol_infrastructure/applications/edxapp/secrets_builder.py

```diff
@@ -0,0 +1,419 @@
+    if proctortrack_url:
+        secrets["PROCTORING_BACKENDS"]["proctortrack"] = {
+            "client_id": '{{ get .Secrets "proctortrack_client_id" }}',
+            "client_secret": '{{ get .Secrets "proctortrack_client_secret" }}',
+            "base_url": proctortrack_url,
+        }
+        secrets["PROCTORING_USER_OBFUSCATION_KEY"] = (
+            '{{ get .Secrets "proctortrack_user_obfuscation_key" }}'
+        )
+
+    # Always include null backend
+    secrets["PROCTORING_BACKENDS"]["null"] = {}
+
+    return secrets
+
+
+def secrets_dict_to_yaml_template(secrets: dict[str, Any]) -> str:
+    """Convert secrets dictionary to Vault YAML template string.
+
+    Args:
+        secrets: Dictionary of secrets configuration
+
+    Returns:
+        YAML string suitable for Vault templating
+    """
+
+    # Custom representer for handling Vault template strings
+    def _str_representer(dumper: Any, data: str) -> Any:
+        if data.startswith("{{") and data.endswith("}}"):
+            # Vault template - use literal block to avoid quoting
+            return dumper.represent_scalar("tag:yaml.org,2002:str", data)
+        return dumper.represent_scalar("tag:yaml.org,2002:str", data)
+
+    # Create custom dumper class
+    class CustomDumper(yaml.Dumper):
+        pass
+
+    # Register custom representer
+    CustomDumper.add_representer(str, _str_representer)
+
+    # Dump to YAML
+    return yaml.dump(
+        secrets, Dumper=CustomDumper, default_flow_style=False, sort_keys=False
+    )
+
+
+def get_database_credentials_template(db_address: str, db_port: int) -> tuple[str, str]:
+    """Generate database credentials template for vault.
+
+    Args:
+        db_address: Database hostname/address
+        db_port: Database port
+
+    Returns:
+        Tuple of (template_string, template_name)
+    """
+    config = {
+        "mysql_creds": {
+            "ENGINE": "django.db.backends.mysql",
+            "HOST": db_address,
+            "PORT": db_port,
+            "USER": '{{ get .Secrets "username" }}',
+            "PASSWORD": '{{ get .Secrets "password" }}',
+        }
+    }
+
+    yaml_template = secrets_dict_to_yaml_template(config)
+    return yaml_template, "00-database-credentials.yaml"
+
+
+def get_database_connections_template(db_address: str, db_port: int) -> tuple[str, str]:
+    """Generate database connections template for vault.
+
+    Args:
+        db_address: Database hostname/address
+        db_port: Database port
+
+    Returns:
+        Tuple of (template_string, template_name)
+    """
+    # Base MySQL credentials
+    mysql_creds = {
+        "ENGINE": "django.db.backends.mysql",
+        "HOST": db_address,
+        "PORT": db_port,
+        "USER": '{{ get .Secrets "username" }}',
+        "PASSWORD": '{{ get .Secrets "password" }}',
+    }
+
+    # Common database options
+    db_options = {
+        "ssl_mode": "REQUIRED",
+        "ssl": {"cipher": "TLSv1.2"},
+        "charset": "utf8mb4",
+    }
```

Comment:
Interestingly, whatever is rendering this seems to create a yaml anchor for this. Not a concern just something I noticed. It looks right to me.